streams/writable-streams/aborting.js

```diff
@@ -204,34 +209,64 @@ promise_test(t => {
 }, 'Aborting a WritableStream puts it in an errored state, with a TypeError as the stored error');
 
 promise_test(t => {
-  const ws = new WritableStream();
+  let resolveWritePromise;
+  const ws = new WritableStream({
+      write() {
+      return new Promise(resolve => {
+        resolveWritePromise = resolve;
+      });
+    }
+  });
   const writer = ws.getWriter();
 
-  const writePromise = promise_rejects(t, new TypeError(), writer.write('a'),
-    'writing should reject with a TypeError');
+  // Perform our first Write to eliminate any race conditions for this test. We'll leave it outstanding until after we abort.
+  writer.write('a');
 
-  writer.abort(error1);
+  // Now perform a second write which will be queued behind this pending write, it should get rejected when we abort.
+  const writePromise = promise_rejects(t, new TypeError(), writer.write('b'), 'writing should reject with a TypeError');
+
+  // FlushAsyncEvents so all implementors will actually call write().
+  return flushAsyncEvents().then(() => {
+      writer.abort(error1);
+      resolveWritePromise();
 
-  return writePromise;
+      return writePromise;
+  });
 }, 'Aborting a WritableStream causes any outstanding write() promises to be rejected with a TypeError');
 
 promise_test(t => {
-  const ws = recordingWritableStream();
+  let resolveWritePromise;
```

Comment:
I believe the race is more in the implementation details, similar to a couple of the other changes above, it only works when async events are queued and not yet processed forcing the intended ordering. However these events could be processed earlier causing a different outcome (since say the write is no longer in-flight or queued).